### PR TITLE
Update flutter-for-react-native.md

### PR DESCRIPTION
**Pull request by wmleler**
Please review and merge these changes (which will merge them into the changes from Geeky Ants, which I just pushed). Then I can finish editing the rest of the document.

### CODE REVIEW

File: flutter-for-react-native.md [hunk 1]
```diff
@@ -265,14 +266,16 @@ import React from "react";
 import { StyleSheet, Text, View } from "react-native";
 {% endprettify %}
 
-Contrary to React Native’s way of importing each component as they are used, in Flutter, you import the `material.dart` from the flutter package, which allows you to use any material design widget without explicitly importing it.
+Contrary to React Native’s way of importing each component needed, in Flutter, you import `material.dart` from the flutter package, which allows you to use any material design widget without explicitly importing it. Dart automatically imports only the widgets that are actually used.
+
+Alternatively, you could import the `cupertino` widgets, the basic `widgets` or another widget library.
```
Comment (Sfshaza) [inline flutter-for-react-native.md]: A good place to add "or a custom widget library"?

File: flutter-for-react-native.md [hunk 2]
```diff
@@ -186,16 +188,15 @@ Whereas Dart uses [`Future`](https://www.dartlang.org/tutorials/language/futures
...
+  HttpRequest.request(url).then((value) {
       print(JSON.decode(value.responseText)['origin']);
   }).catchError((error) => print(error));
 }
 {% endprettify %}
 
 Try it out in [DartPad](https://dartpad.dartlang.org/b68eb981456c5eec03daa3c05ee59486).
 
-**Note:** Check [here](https://www.dartlang.org/tutorials/language/futures) for more details on future.
+**Note:** Check [here](https://www.dartlang.org/tutorials/language/futures) for more details on Futures.
```
Comment (Sfshaza) [inline flutter-for-react-native.md]: Linking on "here" is a huge no-no. I'd also put this  into an <aside>, but maybe leave that for me later. But, for now:

For more information, see the [Futures](xxx) documentation.

**Comment by Sfshaza** [comment]
LGTM (with a few minor comments)

File: flutter-for-react-native.md [hunk 3]
```diff
@@ -79,7 +81,7 @@ Try it out in [DartPad](https://dartpad.dartlang.org/3f4625c16e05eec396d60468837
 
 #### Default value
 
-In Dart, uninitialized variables have an initial value of `null`. Even variables with numeric types are initially null because numbers are objects in Dart. But in JavaScript, variables are `undefined`.
+In Dart, uninitialized variables have an initial value of `null`. Even variables with numeric types are initially null because numbers are objects in Dart. But in JavaScript, uninitialized variables are `undefined`.
```
[the remark this answers is not in the document]
Comment (wmleler) [reply flutter-for-react-native.md]: This is not about Dart, it is about JavaScript. But I'll change it anyway.

File: flutter-for-react-native.md (hunk 2 above)
Comment (wmleler) [reply flutter-for-react-native.md]: There were a TON of these (linking on "here"). I fixed them all.

**Pull request merged**